django_admin_lightweight_date_hierarchy/templatetags/admin_list.py

```diff
@@ -15,6 +15,45 @@ def get_today():
     return datetime.date.today()
 
 
+def default_date_hierarchy_drilldown(year_lookup=None, month_lookup=None):
+    """Generate default drill-down for any level of the hierarchy.
+
+    year_lookup (int or None):
+        Year lookup.
+        None when no lookup.
+    month_lookup (int or None):
+        Month lookup.
+        None when year lookup.
+
+    Returns (generator<datetime.date>):
+        Dates to drill-down on.
+    """
+    if year_lookup is None and month_lookup is None:
+        # Three years from each direction.
+        today = get_today()
+        return (
+            datetime.date(y, 1, 1)
+            for y in range(today.year - 3, today.year + 3 + 1)
+        )
+
+    elif year_lookup is not None and month_lookup is None:
+        # All months of selected year.
+        return (
+            datetime.date(int(year_lookup), month, 1)
+            for month in range(1, 13)
+        )
+
+    elif year_lookup is not None and month_lookup is not None:
+        # All days of month.
+        return (
+            datetime.date(year_lookup, month_lookup, 1) + datetime.timedelta(days=i)
+            for i in range(calendar.monthrange(year_lookup, month_lookup)[1])
+        )
+
+    else:
+        assert 'date hierarchy drilldown makes no sense.'
```

Comment:
Would be better to raise something, then to cause 500. User can pass `year=None, month=!None` after all(?).

Reply:
This scenario shouldn't really happen (this is why I assert).

Two options

1. Follow django's step and handle no filter in else - this way there is no error so no need to raise or assert.
2. Raise something like `ProgrammingError`... imported from django.db so probably not appropriate here.

Reply:
If the assert cannot be triggered by user input, then OK.

Reply:
Well, user can always edit the URl manually but i guess django has validations for that in case of date hierarchy.